handlejoin.go

```diff
@@ -0,0 +1,278 @@
+package gomatrixserverlib
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/matrix-org/gomatrixserverlib/spec"
+	"github.com/matrix-org/util"
+)
+
+type HandleMakeJoinInput struct {
+	Context            context.Context
+	UserID             spec.UserID
+	RoomID             spec.RoomID
+	RoomVersion        RoomVersion
+	RemoteVersions     []RoomVersion
+	RequestOrigin      spec.ServerName
+	RequestDestination spec.ServerName
+	LocalServerName    spec.ServerName
+	LocalServerInRoom  bool
```

Comment:
This doesn't make sense to me. It's never valid for this to be `false`, so why have this at all? I don't really know what the point of this flag is if it cannot handle both true/false? You can see this in the tests which are constantly forced into doing `LocalServerInRoom: true,`